src/java.base/share/classes/sun/net/www/protocol/http/DigestAuthentication.java

```diff
@@ -339,7 +343,11 @@ public String getHeaderValue(URL url, String method) {
      * @return the value of the HTTP header this authentication wants set
      */
     String getHeaderValue(String requestURI, String method) {
-        return getHeaderValueImpl(requestURI, method);
+        try {
+            return getHeaderValueImpl(requestURI, method);
+        } catch (IOException _) {
+            return null;
+        }
     }
```

Comment:
I was wondering if we should let the exception propagate in these cases as well. Though I think these methods get called primarily through pre-emptive authentication, which is less likely to fail (with an exception). So, maybe they are okay...